validation-update.yaml

```diff
@@ -0,0 +1,147 @@
+---
+- name: Run powershell script
+  hosts: windows
+  gather_facts: true
+
+  tasks:
+    - block:
+        - name: 1.Save the result from diag
+          win_command: dcdiag
+          ignore_errors: yes
+          register: dcdiag_output
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: "{{ dcdiag_output.stderr_lines }}"
+            verbosity: 2
+          when:
+            - ansible_host == 'domain_controller'
+
+        - name: 2.Save the result from Replication Status
+          win_shell: powershell.exe Get-Date
+          register: repl_out
+          ignore_errors: yes
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: repl_out
+            verbosity: 2
+
+        - name: 3.Save the result fron nslookup
+          win_command: nslookup 8.8.8.8
+          ignore_errors: yes
+          register: nslookup_test_1
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: "{{ nslookup_test_1 }}"
+          when:
+            - ansible_host == 'domain_controller'
+
+        - name: "nslookup Test 2."
+          ansible.windows.win_shell: nslookup pgcw001dc01
+          become: yes
+          become_user: Administrator
+          register: nslookup_test_2
+          when:
+            - ansible_host == 'domain_controller'
+
+        - debug:
+            msg: "{{ nslookup_test_2 }}"
+          when:
+            - ansible_host == 'domain_controller'
+
+        - name: 4.Save the result from Active leases
+          win_command: powershell.exe Get-Service -Name *dhcp* -ComputerName local
+          register: dhcp_out
+          ignore_errors: yes
+
+        - debug:
+            msg: "{{ dhcp_out }}"
+
+        - name: 5.Save the result from Replication Status
```

Comment:
since we are looking for status of services, what happens is service status is failed

Reply:
in any case dhcp_out will show full logs of that command. if for any reason rc status is not enough to mark a job pass or fail, we can look into  these logs and then add additional step to make it more verbose.